[GitHub] DLS documentation (#1956)
elastic/connectors: docs/document-level-security/GITHUB.md

### Setting up the GitHub connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

## Document level security

> Document level security (DLS) enables you to restrict access to documents based on a user's permissions. This feature is available only for the `Organization`, `Repository Type` in GitHub connector.

Reviewer: ```suggestion
Document level security (DLS) enables you to restrict access to documents based on user permissions. This feature is only available, if the `Repository Type` is set to `Organization` in the GitHub connector.
```
